src/alerting.py

```diff
@@ -0,0 +1,89 @@
+from time import localtime, strftime
+import logging
+import utils
+
+logger = utils.get_logger(__name__)
+
+class AlertManager:
+    """
+    HTTP Packet sniffer and Collector class. Actively listens and filters for HTTP traffic, and adds to a collection
+    separated into requests and responses.
+
+    Attributes:
+        alert_active (bool): Whether an alert is currently active or not.
+        alert_threshold (float): The threshold which has to be crossed on average over a 2 minute span to trigger an alert.
+        This is a float because otherwise with an int rounding down can lead to less tight alert checking.
+        past_alerts (list<tuple(str, str)>): A list of tuples, each containing the string representation of the start
+        and end time of past alerts.
+        current_alert_start_time (str): the string representation of the start time of the current alert, if any.
+    """
+    def __init__(self, threshold):
+        self.alert_active = False
+        self.alert_threshold = float(threshold)
+        self.past_alerts = []
+        self.current_alert_start_time = ''
+        self.alert_logfile = '/var/log/past_alerts.log'
+        self.alert_logger = self.get_alert_logger()
+
+    def get_alert_logger(self):
+        """
+        :return: (logger) A logger specifically for past alert logging.
+        Logs all past alerts into a logfile. This allows for the console program to keep tidy and only show the last 5
+        alerts, while still maintaining record of all past alerts.
+        """
+        logger = logging.getLogger('past_alerts')
+        logger.setLevel(logging.INFO)
+        handler = logging.FileHandler(self.alert_logfile)
+        handler.setLevel(logging.INFO)
+        logger.addHandler(handler)
+        return logger
+
+    def show_alerts(self):
+        if self.alert_active:
+            print(f'High traffic alert currently active! triggered at: {self.current_alert_start_time}.\n')
+        else:
+            print('No currently active alerts.\n')
+        print(self.show_past_alerts())
+
+    def show_past_alerts(self):
+        """
+        :return: (str) String of past alerts.
+        Shows past alert start and end times. If there are more than 5, directs user to look at a logfile.
+        """
+        if not self.past_alerts:
+            return 'No Past alerts to display.\n'
+        past_alerts_str = 'Past Alerts:\n'
+        if len(self.past_alerts) > 5:
```

Comment:
Can remove duplication by creating boolean for threshold condition, then use it to determine the collection of alerts to traverse, and then use it to determine if the extra log message must be logged.

Something like:
```
threshold = len(self.past_alerts) > 5
alerts = self.past_alerts[::-1][:5] if threshold else self.past_alerts[::-1]
for alert_start_time, alert_end_time in alerts:
  past_alerts_str += f'\tPast alert started: {alert_start_time}, ended: {alert_end_time}.\n'
if threshold
  extra_alert
```